types/state.go

```diff
@@ -91,8 +91,14 @@ func NewFromGenesisDoc(genDoc *types.GenesisDoc) (State, error) {
 
 		ConsensusParams:                  *genDoc.ConsensusParams,
 		LastHeightConsensusParamsChanged: genDoc.InitialHeight,
+
+		BaseHeight: 0,
 	}
 	copy(s.AppHash[:], genDoc.AppHash)
 
 	return s, nil
 }
+
+func (s *State) IsGenesis() bool {
+	return s.LastBlockHeight == 0
+}
```

Comment:
this is different from the old calculation right? What's the reason?

Reply:
old calculation supported only initial height of `0`.
to support genesis height > 0, we need other parameter to recognize genesis block